app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,44 @@
+class CommentsController < ApplicationController
+  
+  before_action :logged_in_user, only: [:create]
+
+  def new
+  end
+
+  def create
+    @parent_post = Post.find(params[:post_id])
+
+    if params[:comment_id].present?
+      parent_comment = Comment.find(params[:comment_id])
+      new_comment = parent_comment.comments.build(comment_params)
+      if new_comment.valid? 
+        new_comment.update_attributes(commentable_type: "Comment")
+        create_comment new_comment
+      else 
+        redirect_to @parent_post
+        flash[:danger] = "Your comment may not be empty."
+      end 
+    else
+      new_comment = @parent_post.comments.build(comment_params)
+      if new_comment.valid? 
+        new_comment.update_attributes(commentable_type: "Post")
+        create_comment new_comment
```

Comment:
you can clean this (and it's twin 10 lines above) up a good amount:

- you don't need to set the `commentable_type` attribute, since you're using `@parent_post.comments.build` the commentable type and id will already be set
- I would suggest merging in the `user_id` attribute into the body of the `comment_params` method
- If the `user_id` is part of `comment_params`, then you don't have to check if it's `valid?` before calling update_attributes a couple times, you can just run with `if new_comment.update_attributes(comment_params)`
- It might even be possible to define a `commentable` attribute in comment_params and get this `create` method down to 5 or 6 lines.